feathr_project/feathr/feature.py

```diff
@@ -45,6 +53,27 @@ def __init__(self,
         # self.key could be null, when getting features from registry.
         self.key_alias = [k.key_column_alias for k in self.key if k]
 
+    @classmethod
+    def validate_feature_name(cls, feature_name: str) -> bool:
+        """
+        Only alphabet, numbers, and '_' are allowed in the name.
+                It can not start with numbers. Note that '.' is NOT ALLOWED!
+        """
+        if not feature_name:
+            raise FeatureNameValidationError('Feature name rule violation: empty feature name detected')
+
+        if str.isnumeric(feature_name[0]):
+            raise FeatureNameValidationError(
+                f'Feature name rule violation: feature name cannot start with a number; name={feature_name}')
+
+        feature_validator = re.compile(r'^[a-zA-Z0-9_]+$')
+
+        if not feature_validator.match(feature_name):
+            raise FeatureNameValidationError(
+                f'Feature name rule violation: only letters, numbers, and underscores are allowed in the name; name={feature_name}')
```

Comment:
could you try to combine them together and add a documentation for the regex. Let me know if you need help to come up with a regex. I find this useful when i figure out my regex: https://www.regextester.com/

Reply:
done